File: 2021TOPC/sample_solution/b.py
N = 1000000
table = [1] * (N+1)
table[0] = table[1] = 0
for i in range(2, N+1):
    for j in range(i + i, N+1, i):
        table[j] += i

T = int(input())
for n in map(int, input().split()):
    if table[n] == n:
        print('perfect')
    elif table[n] < n:
        print('deficient')
    else:
        print('abundant')

File: 2021TOPC/sample_solution/b.cpp
#include<bits/stdc++.h>

using namespace std;

int main() {
    vector<int> count(1000000, 1);
    count[0] = count[1] = 0;
    // the following loops: O(nlogn)
    for (int i = 2; i <= 1000000; i++)
        for (int j = i + i; j <= 1000000; j+=i)
            count[j] += i;
    int ncases;
    scanf("%d", &ncases);
    while (ncases--) {
        int x;
        scanf("%d", &x);
        if (count[x] == x)
            puts("perfect");
        else if (count[x] < x)
            puts("deficient");
        else puts("abundant");
    }
}
